Build time error: Can't resolve @swc/helpers/...

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
> ./node_modules/.bin/next info

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.15.0
      npm: 9.6.4
      Yarn: N/A
      pnpm: 8.2.0
    Relevant packages:
      next: 13.3.1-canary.16
      eslint-config-next: 13.3.1-canary.16
      react: 18.2.0
      react-dom: 18.2.0
```


### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://github.com/jianliao/nextjs-rsp-bug

### To Reproduce

1. Git clone https://github.com/jianliao/nextjs-rsp-bug
2. npm i
3. npm run build

The issue is exactly same as #38192.

### Describe the Bug

After upgrade to `next@13.3.1-canary.16`, next build will fail with below error messages:

```
> nextjs-rsp-table@0.1.0 build
> next build

warn  - You have enabled experimental feature (appDir) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback

Failed to compile.

./node_modules/@internationalized/date/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_class_private_field_init.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@react-spectrum/calendar/dist/import.mjs
./node_modules/@adobe/react-spectrum/dist/import.mjs
./app/page.tsx

./node_modules/@internationalized/date/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@react-spectrum/calendar/dist/import.mjs
./node_modules/@adobe/react-spectrum/dist/import.mjs
./app/page.tsx

./node_modules/@react-aria/focus/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@react-spectrum/actiongroup/dist/import.mjs
./node_modules/@adobe/react-spectrum/dist/import.mjs
./app/page.tsx

./node_modules/@react-stately/collections/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@adobe/react-spectrum/dist/import.mjs
./app/page.tsx

./node_modules/@react-stately/grid/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@react-stately/table/dist/import.mjs
./node_modules/@react-spectrum/table/dist/import.mjs
./node_modules/@adobe/react-spectrum/dist/import.mjs
./app/page.tsx


> Build failed because of webpack errors
info  - Creating an optimized production build .%
```

### Expected Behavior

Build success.

### Which browser are you using? (if relevant)

Chrome 113.0.5672.35

### How are you deploying your application? (if relevant)

next build

I'm getting the same errors using **Turborepo**. My additional information mimic for the original reports but I also hopped over to a machine to confirm the results and those details are below. So I've received the same result on both MacOS and Windows. I, too am starting from 'getting started' so this Next app is brand new, but been a non-starter as this presents itself right out the gate.

Thanks and let me know if I can provide any better detail. I can post my macOS details as needed but they mimic the og post as mentioned earlier.


```
> prodkt-builder@0.1.0 build C:\ProdktDevelopment\prodkt-cloud\apps\prodkt-builder
> next build

info  - Skipping linting
info  - Checking validity of types...
info  - Creating an optimized production build...
Failed to compile.

../../node_modules/@internationalized/date/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_class_private_field_init.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../../node_modules/@react-aria/i18n/dist/real-module.mjs
../../node_modules/@react-aria/i18n/dist/import.mjs
../../node_modules/@react-aria/select/dist/import.mjs
../../node_modules/@plasmicapp/react-web/dist/react-web.esm.js
./pages/_app.tsx

../../node_modules/@internationalized/date/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../../node_modules/@react-aria/i18n/dist/real-module.mjs
../../node_modules/@react-aria/i18n/dist/import.mjs
../../node_modules/@react-aria/select/dist/import.mjs
../../node_modules/@plasmicapp/react-web/dist/react-web.esm.js
./pages/_app.tsx

../../node_modules/@react-aria/focus/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../../node_modules/@plasmicapp/react-web/dist/react-web.esm.js
./pages/_app.tsx

../../node_modules/@react-stately/collections/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../../node_modules/@plasmicapp/react-web/dist/react-web.esm.js
./pages/_app.tsx

../../node_modules/@react-stately/list/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/src/_define_property.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../../node_modules/@react-stately/select/dist/import.mjs
../../node_modules/@plasmicapp/react-web/dist/react-web.esm.js
./pages/_app.tsx


> Build failed because of webpack errors
 ELIFECYCLE  Command failed with exit code 1.

```



```
    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 16.20.0
      npm: N/A
      Yarn: N/A
      pnpm: 8.1.1
    Relevant packages:
      next: 13.3.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

I tried all solutions but the same issue! did anyone resolve it?

`error - ./node_modules/@internationalized/date/dist/import.mjs:1:0
Module not found: Can't resolve '@swc/helpers/src/_class_private_field_init.mjs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@react-aria/i18n/dist/real-module.js
./node_modules/@react-aria/i18n/dist/module.js
./node_modules/@nextui-org/react/esm/index.js
./src/providers/index.tsx
<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: ENOENT: no such file or directory, rename 'xx/.next/cache/webpack/client-development-fallback/0.pack_' -> 'xx/.next/cache/webpack/client-development-fallback/0.pack'
`

--- a/package.json
+++ b/package.json
@@ -77,7 +77,7 @@
     "@svgr/webpack": "5.5.0",
     "@swc/cli": "0.1.55",
     "@swc/core": "1.3.51",
-    "@swc/helpers": "0.5.0",
+    "@swc/helpers": "0.5.1",
     "@testing-library/react": "13.0.0",
     "@types/cheerio": "0.22.16",
     "@types/fs-extra": "8.1.0",

--- a/packages/next/package.json
+++ b/packages/next/package.json
@@ -82,7 +82,7 @@
   },
   "dependencies": {
     "@next/env": "13.3.2-canary.5",
-    "@swc/helpers": "0.5.0",
+    "@swc/helpers": "0.5.1",
     "busboy": "1.6.0",
     "caniuse-lite": "^1.0.30001406",
     "postcss": "8.4.14",

